sdk/openai/Azure.AI.OpenAI/src/Custom/AudioTranscription.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI
+{
+    /// <summary> Transcription response. </summary>
+    public partial class AudioTranscription
+    {
+        /// <summary> Initializes a new instance of AudioTranscription. </summary>
+        /// <param name="text"> Transcribed text. </param>
+        /// <param name="task"> Audio transcription task. </param>
+        /// <param name="language"> Language detected in the source audio file. </param>
+        /// <param name="duration"> Duration. </param>
+        /// <param name="segments"> Segments. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="text"/> is null. </exception>
+        internal AudioTranscription(string text, AudioTranscriptionTask? task, string language, TimeSpan duration, IReadOnlyList<AudioTranscriptionSegment> segments)
+        {
+            Argument.AssertNotNull(text, nameof(text));
+
+            Text = text;
+            Task = task;
+            Language = language;
+            Duration = duration;
+            Segments = segments;
+        }
+
+        internal AudioTranscriptionTask? Task { get; }
+
+        /// <summary> Language detected in the source audio file. </summary>
+        public string Language { get; }
+        /// <summary> Duration. </summary>
+        public TimeSpan Duration { get; }
+        /// <summary> Segments. </summary>
+        public IReadOnlyList<AudioTranscriptionSegment> Segments { get; }
```

Comment:
If we were designing the initial API, I'd wholeheartedly agree with you. Exposing all of this information at the top level doesn't seem like it optimizes conceptual disclosure, and redirecting into a details type would make good perfect sense.

I worry it might be just a bit too much of a deviation from the source material to move things around, though; we're already stretching things a bit with intelligent renames (evidenced by feedback) and I can easily envision a developer complaining that we added annoying, extra complexity that made it harder to translate Python/REST into C#.

I'd like to file a work item for us to holistically consider how much movement we can do across object models -- if we reach the conclusion that it's an acceptable cost in conceptual deviation, then I think it'd apply equally to quite a few fields in the older surfaces, too -- especially if we'd apply the lens to all of the knobs in CompletionsOptions and the like.